### ISSUE TITLE
Vurder couchbase, mysql og mongodb som database

### ISSUE THREAD

**Issue by kennethlynne** [EdtechFoundry/edtech-seed-project-api#11]
Krav:
- TBD

**Comment by kennethlynne** [comment]
Vi kan få til søk med elastic search relativt uavhengig av hva vi velger som database (pre-save hooks som populerer elasticsearch indexen etc), så der ryker en av de største argumentene for å bruke couchbase. Couchbase har heller ingen sails adapter.

MongoDb + MySQL then, og ES indeksering as needed f.eks?

**Comment by kennethlynne** [comment]
Waterline håndterer jo (half-assed?) relasjonene på kryss av databasene i grunn.

**Comment by kennethlynne** [comment]
Migrering, solved?! 

https://www.npmjs.com/package/sails-db-migrate
https://www.npmjs.com/package/sails-migrations

**Comment by AlmaasAre** [comment]
Yeah, men når prototype er i produksjon og i bruk, så blir bør vi få opp migrasjonsrutiner. Dette slipper vi med NoSQL. Om vi ikke endrer skjema totalt dat is. Og ja, sails migrate funker forsåvidt det, men det blir en større jobb med migrering uansett. Backup av database, prøve å kjøre migrering, og restore database igjen ser jeg for meg det blir... Hm. men stemmer litt for MongoDB i starten. Da slipper man å i det hele tatt fokusere på migrering osv. Det bare funker, selv om du legger til/fjerner properties/felter/tabeller.

Får jo til det vi vil uansett hvilken DB vi velger, så for vårt use-case er det vanskelig å komme med gode argumenter for den ene eller den andre :p

**Comment by AlmaasAre** [comment]
Har sett noen issues på GH der man ikke får til å joine tabeller fra to forskjellige databaser, så det kan nok være litt half assed :D
